src/nni_manager/training_service/remote_machine/remoteMachineTrainingService.ts

```diff
@@ -320,9 +324,21 @@ class RemoteMachineTrainingService implements TrainingService {
                     throw new Error(`codeDir ${remoteMachineTrailConfig.codeDir} is not a directory`);
                 }
 
-                // Validate to make sure codeDir doesn't have too many files
                 try {
+                    // Validate to make sure codeDir doesn't have too many files
                     await validateCodeDir(remoteMachineTrailConfig.codeDir);
+                    // Copy codeDir to remote machine
+                    for (const [rmMeta, executorManager] of this.machineExecutorManagerMap.entries()) {
+                        const executor: ShellExecutor = await executorManager.getAvailableExecutor();
+                        if (executor !== undefined) {
+                            await executor.createFolder(this.remoteExpCodeDir);
```

Comment:
This line can also put in async promise.

Reply:
fixed.